py/autoevals/moderation.py

```diff
@@ -0,0 +1,94 @@
+import threading
+
+from braintrust_core.score import Score, Scorer
+
+from .oai import arun_cached_request, run_cached_request
+
+REQUEST_TYPE = "moderation"
+
+
+class Moderation(Scorer):
+    """
+    A scorer that uses OpenAI's moderation API to determine if AI response contains ANY flagged content.
+    """
+
+    _CACHE = {}
+    _CACHE_LOCK = threading.Lock()
+
+    threshold = None
+    extra_args = {}
+
+    def __init__(self, threshold=None, api_key=None, base_url=None):
+        """
+        Create a new Moderation scorer.
+
+        :param threshold: A prefix to prepend to the prompt. This is useful for specifying the domain of the inputs.
+        :param api_key: OpenAI key
+        :param base_url: Base URL to be used to reach OpenAI moderation endpoint.
+        """
+        self.threshold = threshold
+
+        self.extra_args = {}
+
+        if api_key:
+            self.extra_args["api_key"] = api_key
+        if base_url:
+            self.extra_args["base_url"] = base_url
+
+    async def _a_moderation(self, value):
+        with self._CACHE_LOCK:
+            if value in self._CACHE:
+                return self._CACHE[value]
```

Comment:
Why do you have this cache? Is it mirroring what we do for embeddings?

If so, the reason we do that is that people tend to use embeddings in a pairwise manner (eg via the new `ListContains`), which may be an unnecessary optimization.

Reply:
Oh nice. Didn't realize it. I removed the cache - The class should be alot simpler now.